apps/api/src/app/workflows-v2/workflow.controller.ts

```diff
@@ -114,9 +114,18 @@ export class WorkflowController {
   @UseGuards(UserAuthGuard)
   async getWorkflow(
     @UserSession(ParseSlugEnvironmentIdPipe) user: UserSessionData,
-    @Param('workflowId', ParseSlugIdPipe) workflowId: IdentifierOrInternalId
+    @Param('workflowId', ParseSlugIdPipe) workflowId: IdentifierOrInternalId,
+    @Query('environmentId') environmentId?: string
```

Comment:
I added this as an optional param to be able to retrieve workflow from different than current user session environment.

We use this when deciding if we need to show confirmation modal informing that workflow already exist in production - we need to find dev `workflowId` in prod environment.